go/runtime/client/client.go

```diff
@@ -561,8 +550,8 @@ func (c *runtimeClient) Start() error {
 // Implements service.BackgroundService.
 func (c *runtimeClient) Stop() {
 	// Watchers.
-	for _, watcher := range c.watchers {
-		watcher.Stop()
+	for _, submitter := range c.txSubmitters {
```

Comment:
Just noticed that this loop (and below) should probably be under a `c.Lock()` guard as in theory a request could come in right when this is shutting down?

Reply:
nice find! yeah indeed should be, will update.